chain/sync.go

```diff
@@ -267,6 +267,7 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		return err
 	}
 	if !semverCompatible(requiredAPIVersion, api.Version) {
+		client.Close()
```

Comment:
good catch that we need to close this, but let's instead do it with a defer so it happens on all error paths

```
client, err := wsrpc.Dial(...)
if err != nil {
  // ...
}
defer client.Close()
```